Sources/Swift/Persistence/SentryScopePersistentStore.swift

```diff
@@ -0,0 +1,222 @@
+@_implementationOnly import _SentryPrivate
+
+@objc
+enum SentryScopeField: UInt, CaseIterable {
+    case context
+    case user
+    
+    var name: String {
+        switch self {
+        case .context:
+            return "context"
+        case .user:
+            return "user"
+        }
+    }
+}
+
+@_spi(Private) @objc public protocol SentryFileManagerProtocol {
+    func moveState(_ stateFilePath: String, toPreviousState previousStateFilePath: String)
+    func readData(fromPath path: String) throws -> Data
+    @objc(writeData:toPath:)
+    @discardableResult func write(_ data: Data, toPath path: String) -> Bool
+    func removeFile(atPath path: String)
+    func getSentryPathAsURL() -> URL
+}
+
+@objc
+@_spi(Private) public class SentryScopePersistentStore: NSObject {
```

Comment:
`l`: To be honest, this file is already growing massively. Having multiple store types as subclasses or wrappers of the scope persistent store could be a way of sharing the generic persistence logic, while having simpler files/types.

No strong opinion here, so proceed as you believe best.

Reply:
This is a trade off, there is no clear solution either, but I think this approach is easier to maintain.